packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -498,11 +518,91 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
         });
     }
 
+    private getCustomUserMentionExtension(): Node<MentionOptions> {
+        return Mention.extend({
+            parseHTML() {
+                return [
+                    {
+                        tag: richTextMentionUsersViewTag
+                    }
+                ];
+            },
+            addAttributes() {
+                return {
+                    href: {
+                        default: null,
+                        parseHTML: element => element.getAttribute('mention-href'),
+                        renderHTML: attributes => {
+                            if (!attributes.href) {
+                                return {};
+                            }
+                            return {
+                                'mention-href': attributes.href as string
+                            };
+                        }
+                    },
+
+                    label: {
+                        default: null,
+                        parseHTML: element => element.getAttribute('mention-label'),
+                        renderHTML: attributes => {
+                            if (!attributes.label) {
+                                return {};
+                            }
+
+                            return {
+                                'mention-label': attributes.label as string
+                            };
+                        }
+                    }
+                };
+            },
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            renderHTML({ HTMLAttributes }) {
+                return [
+                    richTextMentionUsersViewTag,
+                    mergeAttributes(
+                        this.options.HTMLAttributes,
+                        HTMLAttributes,
+                        { 'disable-editing': true }
+                    )
+                ];
+            }
+        }).configure({
```

Comment:
Moved the link configuration inside `getCustomLinkExtension` method. As for mention extension, there might be different configuration for other mentions in future, so keeping it inside to avoid confusion while setting it to tip tap editor.